src/hb-subset-map.cc

```diff
@@ -0,0 +1,54 @@
+#include <new>
+#include <unordered_map>
+
+#include "hb-private.hh"
+#include "hb-object-private.hh"
+
+#include "hb-subset-map.hh"
+
+struct hb_map_t {
+  hb_object_header_t header;
+  ASSERT_POD();
+  std::unordered_map<hb_codepoint_t, hb_codepoint_t> *map;
+};
+
+hb_map_t *
+hb_map_create_or_fail ()
+{
+  hb_map_t *map = hb_object_create<hb_map_t> ();
+  if (map)
+  {
+    map->map = new (std::nothrow) std::unordered_map<hb_codepoint_t, hb_codepoint_t>();
+  }
+  if (!map->map)
```

Comment:
This is invalid access if first map allocation failed. Instead, just return fast if allocation failed.

Reply:
Fixed.